packages/domain/lib/usecases/facebook_auth_usecase.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:domain/entities/user_entity.dart';
+import 'package:domain/repositories/auth_repository.dart';
+import 'package:domain/usecases/usecase.dart';
+
+class FacebookAuthUseCase implements UseCase<Future<UserEntity?>> {
+  final AuthRepository _authService;
+
+  FacebookAuthUseCase(this._authService);
+
+  @override
+  Future<UserEntity?> call() async => await _authService.authWithFacebook();
```

Comment:
in case of throwing exception from service, UserEntity will always be non-nullable.
up to u - await seems to be redundant. u will call await in bloc

Reply:
Fixed.